src/org/rascalmpl/repl/TerminalProgressBarMonitor.java

```diff
@@ -395,6 +493,21 @@ private ProgressBar findBarByName(String name) {
             .filter(b -> b.threadId == Thread.currentThread().getId())
             .filter(b -> b.name.equals(name)).findFirst().orElseGet(() -> null);
     }
+
+    private UnfinishedLine findUnfinishedLine() {
+        UnfinishedLine before = unfinishedLines.stream()
+            .filter(l -> l.threadId == Thread.currentThread().getId())
+            .findAny()
+            .orElse(null);
+
+        if (before == null) {
+            UnfinishedLine l = new UnfinishedLine();
```

Comment:
This could be replaced by `orElseGet` function?

Reply:
I don't know how because I also need the side-effect in the list. The caffeine API does this better.

Reply:
```
() -> {
//Side effects.
return x;
}
```